### PR TITLE
fix: Remove references to deprecated AWS Lambda runtimes

### PR DESCRIPTION
**Pull request by bartelemi**
what.

- Remove references to deprecated AWS Lambda runtimes (Python 2.7, 3.6 and Node 10.x).
- Update tests, documentation, etc.

why.

Since AWS is no longer supporting these runtimes it would be better if the documentation was up to date and pointed users towards new/supported configuration that actually works.

references.

Based on [AWS docs](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html#runtime-support-policy).

Test results (local run, excluding Ruby):
```text
  2823 passing (2m)
  179 pending

 Notice: Some tests were skipped due to following environment issues:

 - test/unit/lib/plugins/aws/invokeLocal/index.test.js: Ruby: "before all" hook in "Ruby" (in: t/unit/lib/plugins/aws/invoke-local/index.test.js)

   Ruby runtime is not installed
```

### CODE REVIEW

File: docs/providers/tencent/cli-reference/create.md
Comment (GrahamCampbell) [inline docs/providers/tencent/cli-reference/create.md:56]: Probably we should use Python 3.9 here, like in the other places.

File: test/unit/lib/cli/interactive-setup/dashboard-set-org.test.js
Comment (GrahamCampbell) [inline test/unit/lib/cli/interactive-setup/dashboard-set-org.test.js:46]: We can't add Python 3.10 here until after https://github.com/serverless/serverless/pull/11922 is merged.

File: test/unit/lib/cli/interactive-setup/dashboard-login.test.js
Comment (GrahamCampbell) [inline test/unit/lib/cli/interactive-setup/dashboard-login.test.js:31]: We can't add Python 3.10 here until after https://github.com/serverless/serverless/pull/11922 is merged.

File: docs/providers/tencent/cli-reference/create.md
Comment (GrahamCampbell) [reply docs/providers/tencent/cli-reference/create.md:56]: In any case, we can't add Python 3.10 here until after https://github.com/serverless/serverless/pull/11922 is merged.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: docs/providers/tencent/cli-reference/create.md
Comment (bartelemi) [reply docs/providers/tencent/cli-reference/create.md:56]: Good catch, I will change it to Python 3.9.

I will keep the other Python3.10 changes thought and I suggest we wait until that PR is merged. I will make sure to resolve any conflicts swiftly.

**Review by medikoo — CHANGES_REQUESTED** [review]
@bartelemi great thanks for that update. Can you rebase with `main` (as some important changes to list of runtimes were merged recently)

**Comment by bartelemi** [comment]
> @bartelemi great thanks for that update. Can you rebase with `main` (as some important changes to list of runtimes were merged recently)

Rebased :)

**Review by medikoo — CHANGES_REQUESTED** [review]
@bartelemi great thanks! It looks really good, just one problem (as picked by the CI) is that some of the changes do not follow Prettier formatting. 

Can you update so it's ensured?

**Comment by bartelemi** [comment]
Prettier warning fix, pls rerun workflows.

**Review by medikoo — APPROVED** [review]
Thank you @bartelemi

**Pull request merged**
